castle/cms/upgrades/__init__.py

```diff
@@ -45,7 +45,11 @@ def upgrade(context, logger=None):
 upgrade_3001 = default_upgrade_factory('3001')
 upgrade_3003 = default_upgrade_factory('3003')
 upgrade_3004 = default_upgrade_factory('3004')
-upgrade_3005 = default_upgrade_factory('3005')
+
+
+def upgrade_3005(site, logger=None):
+    # add in princexml username and password
+    re_register_profile(IAPISettings, 'castle')
```

Comment:
why the change here for 3005 ?

Reply:
it looks like it was originally a mistake to me, and i was retroactively attempting to correct it (if, for no other reason, it is not used as an incorrect template by another developer down the line). there is no profile for 3005, and the only thing the upgrade step is supposed to do is add those registry values, so that requires a re-register of an interface (crap, i named that function wrong - should be re_register_interface, not re_register_profile), not a standard upgrade function

Reply:
alrighty, sounds good